### PR TITLE
Added task for forms module in React

### PR DESCRIPTION
**Pull request by kravaring**
#### 🤔 This is a ...

- [x] 🌟 New task
- [ ] 🌐 New module
- [ ] ⚙️ Update to an existing task
- [ ] 🔧 Update to an existing module
- [ ] 🔗 Update or addition of external resources or links
- [ ] 🐛 Fix in a task or related content
- [ ] 🛠 Fix in a module or related content
- [ ] ✏️ Fixed a typo or grammatical error
- [ ] 🔗 Fixed a broken link
- [ ] ❓ Other (specify: **\*\*\*\***\_\_\_\_**\*\*\*\***)

#### Description

- **Brief Overview:**
  Practical task for the Forms module in React

#### Checklist

- [x] ✅ I have performed a self-review of my own code.
- [ ] 📝 I have commented my code, particularly in hard-to-understand areas.
- [x] 🔧 I have made corresponding changes to the documentation (if applicable).
- [x] 🚫 My changes generate no new warnings or errors.

### CODE REVIEW

File: react/modules/tasks/forms.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        - `lint`: For running the lint command.
+        - `format:fix`: For running Prettier's --write command.
+5. Add **React Hook Form**, **Yup**, **Redux Toolkit** and **React Router** to the application.
+
+## Application Requirements
+
+1. Routing. There will be 3 routes:
+   - Main, should have links to other 2 routes
+   - Route for the form created using uncontrolled components approach
+   - Route for the similar form, but created with the help of the **React Hook Form**
```
Comment (Margaryta-Maletz) [inline react/modules/tasks/forms.md:28]:
```suggestion
   - Route for the similar form, but created using **React Hook Form**
```

File: react/modules/tasks/forms.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+### Penalties
+
+- TypeScript isn't used: **-95 points**
+- Usage of _any_: **-20 points per each**
+- Usage of _ts-ignore_: **-20 points per each**
+- Presence of _code-smells_ (God-object, chunks of duplicate code), commented code sections: **-10 points per each**
+- Validation is implemented without **Yup** - **-25 points**
+- Commits after the deadline: **-40 points**
+- Usage of component libraries, e.g. Material UI, Ant Design: **-100 points**
```
Comment (Margaryta-Maletz) [inline react/modules/tasks/forms.md:68]: Maybe better move this point to the first position?

**Review by Margaryta-Maletz — APPROVED** [review]
I added comments because some points were unclear to me.

File: react/modules/tasks/forms.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+   - Route for the similar form, but created with the help of the **React Hook Form**
+2. Redux. Use redux to store the data provided by both approaches on the Main route. You can use tiles to display data taken from each form.
+3. Forms
+   Both forms will collect the same data:
+   - name (validate for first uppercased letter)
+   - age (should be number, no negative values)
+   - email (validate for email)
+   - 2 passwords (should match, display the password strength: 1 number, 1 uppercased letter, 1 lowercased letter, 1 special character)
+   - gender (you can use radio buttons or select control)
+   - accept T&C (checkbox)
```
[the remark this answers is not in the document]
Comment (kravaring) [reply react/modules/tasks/forms.md]: Done

File: react/modules/tasks/forms.md (hunk 1 above)
Comment (kravaring) [reply react/modules/tasks/forms.md]: Done

File: react/modules/tasks/forms.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+   - input control to upload picture (validate size and extension, allow png jpeg, save in redux store as base64)
+   - autocomplete control to select country (all countries should be stored in the Redux store)
+     Form should contain labels, which should be connected with inputs (look at **htmlFor**)
+4. Validation
+   Implement validation according to the inputs description from p. 8. Use **Yup** for validation. Show errors either above each component, or below (but stick with one approach everywhere). Block submitting the form before all the errors are fixed (disable submit button). Good UX assumes that there are no "jumps" when showing errors.
```
[the remark this answers is not in the document]
Comment (kravaring) [reply react/modules/tasks/forms.md]: Remnants of the old task. Fixed.

File: react/modules/tasks/forms.md (hunk 2 above)
Comment (kravaring) [reply react/modules/tasks/forms.md]: Done

**Pull request merged**